geckomat/change_model/applyKcatConstraints.m

```diff
@@ -79,22 +79,37 @@
     model.S(linearIndices) = -newKcats(:,4); %Substrate = negative
     
 else %GECKO light formulation, where prot_pool represents all usages
-    newKcats=zeros(numel(updateRxns),1);
-    for i=1:numel(updateRxns)
-        %TODO:The current code is incorrect, it takes the sum of all
-        %enzymes that are associated. But there could be isoenzymes, so not
-        %all enzymes are necessarily required. Not sure how this can easily
-        %be parsed (without running expandModel).
-        enzymes     = find(model.ec.rxnEnzMat(i,:));
-        MW          = sum(model.ec.mw(enzymes));
-        newKcats(i) = MW/model.ec.kcat(i);
-        %TODO:Correct units.
-    end
-    %All draw directly from protein pool
+    nextIndex = 1;
     prot_pool_idx = find(ismember(model.mets,'prot_pool'));
-
-    [~,rxnIds] = ismember(model.ec.rxns(updateRxns),model.rxns);
-    model.S(prot_pool_idx,rxnIds) = -newKcats; %Substrate = negative
+    for i = 1:numel(model.rxns)
+        if nextIndex <= length(model.ec.rxns) && strcmp(model.rxns(i), model.ec.rxns(nextIndex))
+            updated = false;
+            newMWKcats=[];
+            %Reactions with isozymes will have several rows in ec.rxns etc.
+            %Loop through all reactions, and for each reaction look at all isozymes
+            %Then use the lowest MW/kcat among the isozymes - when optimizing, that
+            %is the one that will be used anyway.
+            while true
+                if updateRxns(nextIndex)
+                    updated = true;
+                end
+                enzymes = find(model.ec.rxnEnzMat(nextIndex,:));
+                MW = sum(model.ec.mw(enzymes).* model.ec.rxnEnzMat(nextIndex,:)); %multiply MW with number of subunits
+                newMWKcats = [newMWKcats MW/model.ec.kcat(nextIndex)]; %Light model: protein usage is MW/kcat
+                
+                nextIndex = nextIndex + 1;
+                if  nextIndex > length(model.ec.rxns) || ~strcmp(model.rxns(i), model.ec.rxns(nextIndex))
+                    break;
+                end
+            end
+            if (updated)
+                %Light model: always use the "cheapest" isozyme, that is what the 
+                %optimization will choose anyway unless isozymes are individually constrained.
+                %TODO:Check that the units are correct.
+                model.S(prot_pool_idx, i) = min(newMWKcats) * 3600; %per second -> per hour
```

Comment:
Agreed, I can fix that

Reply:
Hmm, I'm a bit confused here. So, the protein pool reaction is the same for both light and full, so if we turn this around, which I think makes sense, I had it in that order in the brain modeling for example, we also need to turn this around in the full model. So, there is a sign that needs to be changes somewhere there as well. Could any of you advise which place would be best, I have some problem understanding the consequences?

Reply:
in the normal gecko of this function, enzyme is added as substrates, so here this line should be  `model.S(prot_pool_idx, i) = min(newMWKcats) * (-3600)`; @Yu-sysbio @edkerk either you may need to double check this.

Reply:
> 

It would be better to keep the same structure as GECKO1&2.

Here show reactions in ecYeast (created by GECKO1&2) as an example:
```
prot_pool_exchange	 -> prot_pool[c]
draw_prot_P15454	20.6371 prot_pool[c]  -> prot_P15454[c]
r_0330No1	s_0434[c] + s_0615[c] + 3.0864e-06 prot_P15454[c]  -> s_0394[c] + s_0613[c]
```
In GECKO1&2, `prot_pool` is as the product in **prot_pool_exchange** reaction and as the substrate in **draw_prot_xxxx** reaction. Then the product of the **draw_prot_xxxx** reaction is as the substrate in the corresponding metabolic reaction.

In GECKO3, currently it seems different, which I think should be corrected.
Regarding the light version, `prot_pool` is as the SUBSTRATE in **prot_pool_exchange** reaction and as the PRODUCT in the corresponding metabolic reaction, which is just the opposite (see reactions below).
```
prot_pool_exchange	prot_pool[c] -> 
r_0330	s_0434[c] + s_0615[c] -> s_0394[c] + s_0613[c] + (mw/kcat) prot_pool[c]
```

Regarding the full version, `prot_pool` is as the SUBSTRATE in **prot_pool_exchange** reaction and as the PRODUCT in **draw_prot_xxxx** reaction. However, the substrate in **draw_prot_xxxx** reaction is also the substrate in the corresponding metabolic reaction, which is incorrect (see reactions below).
```
prot_pool_exchange	prot_pool[c] -> 
draw_prot_P15454	20.6371 prot_P15454[c]  -> prot_pool[c]
r_0330No1	s_0434[c] + s_0615[c] + 3.0864e-06 prot_P15454[c]  -> s_0394[c] + s_0613[c]
```

Could consider my suggestions on the correction:
Regarding the light version, in **prot_pool_exchange** reaction change the stoich coeff of `prot_pool` to 1, and in metabolic reactions change `prot_pool` to negative.
Regarding the full version, in **prot_pool_exchange** reaction change the stoich coeff of `prot_pool` to 1, and in **draw_prot_xxxx** reaction change `prot_pool` to negative and `prot_xxxx` to 1 (note that MW should be as the stoich coeff of `prot_pool` rather than `prot_xxxx`, which should also be corrected).

Reply:
@edkerk I think you did this change, right, from version 2 to 3? Did you have a good reason for this, or did it just "happen"? I'm just thinking it would be good to know before I go in and change it.

Reply:
The stoichiometric coefficients in the draw_prot and prot_pool reactions are indeed incorrect in the full version, this is a bug that hadn't been noticed yet, as the GECKO3-derived ec-models haven't yet been tested. Put this will indeed need to be fixed in the relevant functions.

Reply:
Fixed in 44e9ffd for full GECKO, and ce72bb1249f0245fc96f05608deb8a276a4eca83 for GECKO light.